### ISSUE TITLE
Metadata not correctly updated in notifications when PATCH is used

### ISSUE THREAD

**Issue by fgalan** [telefonicaid/fiware-orion#3036]
At .test cases/1156_mq_as_uri_param_for_metadata/q_and_mq_as_uri_param_for_metadata.test:

The following update is done:

```
echo "11b. Update E3, setting M1 to A1"
echo "================================"
payload='{
  "A1": {
    "metadata": {
      "M1": {
        "value": 3
      }
    }
  }
}'
orionCurl --url /v2/entities/E3/attrs --payload "$payload" -X PATCH
```

According to patch entities semantics this operation changes A1 in the following way:

* Value: null
* Type: "None"
* Metadata: *only* M1 with value 3.

Actually, the get entity operation (done just after the update) honours this semantics:

```
11c. GET E3, just to check the value/type of its A1 attribute
=============================================================
HTTP/1.1 200 OK
Content-Length: 106
Content-Type: application/json
Fiware-Correlator: REGEX([0-9a-f\-]{36})
Date: REGEX(.*)

{
    "A1": {
        "metadata": {
            "M1": {
                "type": "Number",
                "value": 3
            }
        },
        "type": "None",
        "value": null
    },
    "id": "E3",
    "type": "Thing"
}
```

but the notification generated by the update doesn't:

```
12. Dump accumulator, see E3
============================
POST http://localhost:REGEX(\d+)/notify
Fiware-Servicepath: /
Content-Length: 194
User-Agent: orion/REGEX(\d+\.\d+\.\d+.*)
Ngsiv2-Attrsformat: normalized
Host: localhost:REGEX(\d+)
Accept: application/json
Content-Type: application/json; charset=utf-8
Fiware-Correlator: REGEX([0-9a-f\-]{36})

{
    "data": [
        {
            "A1": {
                "metadata": {
                    "M1": {
                        "type": "Number", 
                        "value": 3
                    }, 
                    "M2": {
                        "type": "Number", 
                        "value": 3
                    }
                }, 
                "type": "None", 
                "value": null
            }, 
            "id": "E3", 
            "type": "Thing"
        }
    ], 
    "subscriptionId": "REGEX([0-9a-f]{24})"
}
```

The bug is that M2 must not be in the notification.

**Comment by fgalan** [comment]
Could be related with https://github.com/telefonicaid/fiware-orion/issues/2587... Not sure

**Comment by Anjali-NEC** [comment]
Hi @fgalan sir,
As per my investigation, In the test case `11b.Update E3, setting M1 to A1` which is written in file https://github.com/telefonicaid/fiware-orion/blob/master/test/functionalTest/cases/1156_mq_as_uri_param_for_metadata/q_and_mq_as_uri_param_for_metadata.test the expectation is as the patch operation should change M2 metadata to M1 metadata of A1 attribute.
To confirm the updated value, if we execute `11.c GET E3, setting M1 to A1` then result is shown as per the expectation i.e. M1 metadata in A1 attribute.

And if the notification operation is performed i.e `12. Dump accumulator, see E3` below is the output:
```
POST [http://localhost:REGEX(\d+)/notify]http://localhost:REGEX(\d+)/notify
Fiware-Servicepath: /
Content-Length: 161
User-Agent: orion/REGEX(\d+\.\d+\.\d+.*)
Ngsiv2-Attrsformat: normalized
Host: localhost:REGEX(\d+)
Accept: application/json
Content-Type: application/json; charset=utf-8
Fiware-Correlator: REGEX([0-9a-f\-]{36})

{
    "data": [
        {
            "A1": {
                "metadata": {
                    "M1": {
                        "type": "Number",
                        "value": 3
                    }
                },
                "type": "None",
                "value": null
            },
            "id": "E3",
            "type": "Thing"
        }
    ],
    "subscriptionId": "REGEX([0-9a-f]{24})"
}
```

As per my understanding, current behaviour is ok and as per expectation. So there is no need to update.
For your reference please check the attached test file and screenshot. Kindly let me know your opinion.

![3036](https://user-images.githubusercontent.com/52558146/82314200-12cad800-99e7-11ea-9852-883257fccb09.JPG)

**Comment by fgalan** [comment]
The text for the test shown in my original post of this issue seems to be old. I have find the change in this commit https://github.com/telefonicaid/fiware-orion/commit/0722433203bc55e775e3166c5ad708ef8017528d#diff-c802915e2d5e80852f8865fc83091a65, which belongs to PR https://github.com/telefonicaid/fiware-orion/pull/3316, which in sequence is associated to issues #3310 and #3311, solved as part of [release 2.0.0](https://github.com/telefonicaid/fiware-orion/releases/tag/2.0.0) time ago.

In conclusion: the issue is already solved :) I'm closing it.

Thanks for your investigation! It is also important for the project to locate already solved issues so we can properly close them.
